### PR TITLE
Add benchmark for sponge gadget

### LINKED ISSUE #206: Add benchmarks for poseidon sponge
#### Summary
Add benchmarks for the sponge gadget proof creation and verification.

#### Possible solution design or implementation
Same structure as the merkle opening benchmarks.

#### Additional context
N/A

### PR DESCRIPTION
**Pull request by moCello**
Resolves #206

### CODE REVIEW

**Review by ureeves — APPROVED** [review]
LGTM but... could you post the results? 😊

**Comment by moCello** [comment]
> LGTM but... could you post the results?

Sure! I thought the benchmarks are more of a preemptive measurement so we have a way to measure future changes by running them locally before and after the change.

Where would you like to have the results posted? Is it something that you think should be in the README?

File: benches/sponge.rs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    let circuit = SpongeCircuit::new(message);
+
+    // Benchmark sponge native
+    c.bench_function("sponge native", |b| {
+        b.iter(|| {
+            dusk_poseidon::sponge::hash(black_box(&circuit.message));
+        })
+    });
+
+    // Benchmark circuit compilation
```
Comment (xevisalle) [inline benches/sponge.rs:71]: I don't think it is useful to benchmark the compilation, plus, it'll take more time to run the benchmarks.

**Review by xevisalle — APPROVED** [review]
LGTM with a nit commented in the code.

File: benches/sponge.rs (hunk 1 above)
Comment (moCello) [reply benches/sponge.rs:71]: opened issue #209 for this

**Pull request merged**